src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/DateTimeVariantTest.cs

```diff
@@ -200,10 +200,14 @@ private static void TestSqlDataRecordParameterToTVP_Type(object paramValue, stri
             }
             catch (Exception e)
             {
-                if (IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName))
-                    LogMessage(tag, "[EXPECTED EXCEPTION] " + e.Message);
-                else
-                    DisplayError(tag, e);
+                Assert.True(IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName), e.Message);
+                //if (IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName))
+                //    LogMessage(tag, "[EXPECTED EXCEPTION] " + e.Message);
+                //else
+                //    DisplayError(tag, e);
+
+                throw;
```

Comment:
This throw isn't needed. When the `InvalidCastException` is thrown in `TestSimpleParameter_Type()`, `SendInfo()`ends and `TestSqlDataRecordParameterToTVP_Type()` is never run. 

In fact, it looks like the above `Assert.True()` is never even called.

Reply:
At this stage I mainly tried to not touch the logic. We used to get to this part frequently and I just kept it there to remind me later about the approach. I can comment them out and will remove them later when the test is redesigned totaly.